src/ContentBundle/Block/Service/DownloadsBlockService.php

```diff
@@ -53,6 +58,32 @@ public function buildManageForm(FormBuilderInterface $builder, array $options)
     }
 
     /**
+     * Download media item.
+     *
+     * @param int $id
+     *
+     * @return Response
+    */
+    public function downloadMedia($id)
+    {
+        $media = $this->mediaManager->getRepository()->find($id);
+        $filePath = $this->container->getParameter('kernel.root_dir').'/../web/uploads/'.$media->getName();
```

Comment:
A few things to note here;

- This will not work if the file is stored somewhere else than on the local machine (e.g. AWS S3). Please use LiipImagineBundle's cachemanager for this. Like here; https://github.com/Opifer/Cms/blob/master/src/MediaBundle/EventListener/Serializer/MediaEventSubscriber.php#L83.
- You pass $media->getName(), but this is not necessarily the filename. The user might change the name to something else (e.g. "Beautiful picture of lovely landscape"). Use $media->getReference() instead.
- There's no need to inject the whole ServiceContainer only to get a parameter. You can just pass parameters to a service. Read http://symfony.com/doc/current/book/service_container.html#service-parameters